test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs

```diff
@@ -480,6 +367,128 @@ public void EntryPointInputArgsChecks()
+            const int nModels = 5;
+            var splitOutput = CVSplit.Split(Env, new CVSplit.Input { Data = dataView, NumFolds = nModels + 1 });
+            var predictorModels = new IPredictorModel[nModels];
+            var individualScores = new IDataView[nModels];
+            for (int i = 0; i < nModels; i++)
+            {
+                var data = splitOutput.TrainData[i];
+                var lrInput = new LogisticRegression.Arguments
+                {
+                    TrainingData = data,
+                    L1Weight = (Single)0.1 * i,
+                    L2Weight = (Single)0.01 * (1 + i),
+                    NormalizeFeatures = NormalizeOption.No
+                };
+                predictorModels[i] = LogisticRegression.TrainBinary(Env, lrInput).PredictorModel;
+                individualScores[i] =
+                    ScoreModel.Score(Env,
+                        new ScoreModel.Input { Data = splitOutput.TestData[nModels], PredictorModel = predictorModels[i] })
+                        .ScoredData;
+
+                individualScores[i] = new CopyColumnsTransform(Env,
+                    new CopyColumnsTransform.Arguments()
+                    {
+                        Column = new[]
+                        {
+                            new CopyColumnsTransform.Column()
+                            {
+                                Name = MetadataUtils.Const.ScoreValueKind.Score + i,
+                                Source = MetadataUtils.Const.ScoreValueKind.Score
+                            },
+                        }
+                    }, individualScores[i]);
+                individualScores[i] = new DropColumnsTransform(Env,
+                    new DropColumnsTransform.Arguments() { Column = new[] { MetadataUtils.Const.ScoreValueKind.Score } },
+                    individualScores[i]);
+            }
+
+            var avgEnsembleInput = new EnsembleCreator.ClassifierInput { Models = predictorModels, ModelCombiner = EnsembleCreator.ClassifierCombiner.Average };
+            var avgEnsemble = EnsembleCreator.CreateBinaryEnsemble(Env, avgEnsembleInput).PredictorModel;
+            var avgScored =
+                ScoreModel.Score(Env,
+                    new ScoreModel.Input { Data = splitOutput.TestData[nModels], PredictorModel = avgEnsemble }).ScoredData;
+
+            var medEnsembleInput = new EnsembleCreator.ClassifierInput { Models = predictorModels };
+            var medEnsemble = EnsembleCreator.CreateBinaryEnsemble(Env, medEnsembleInput).PredictorModel;
+            var medScored =
+                ScoreModel.Score(Env,
+                new ScoreModel.Input { Data = splitOutput.TestData[nModels], PredictorModel = medEnsemble }).ScoredData;
+
+            var regEnsembleInput = new EnsembleCreator.RegressionInput() { Models = predictorModels };
+            var regEnsemble = EnsembleCreator.CreateRegressionEnsemble(Env, regEnsembleInput).PredictorModel;
```

Comment:
>CreateRegressionEnsemble [](start = 46, length = 24)

Regression ensemble created here while all the underlying models are binary classifier (LogisticRegression). Is it correct?

Reply:
they both create float score field, so it rather create new regression model or reuse binary model.

---
In reply to: [197286978](https://github.com/dotnet/machinelearning/pull/379#discussion_r197286978) [](ancestors = 197286978)